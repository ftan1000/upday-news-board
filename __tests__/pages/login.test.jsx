import React from 'react';
import {shallow} from 'enzyme';

import LoggedInFooter from '../../pages/loggedInFooter';
import Header from '../../components/header';
import Login from '../../pages/login';
import LoginForm from '../../components/loginForm';

describe('Login', () => {

  it('contains expected components', () => {
    const wrapper = shallow(<Login/>);
    expect(wrapper.find(Header)).toBeTruthy();
    expect(wrapper.find(LoginForm)).toBeTruthy();
    expect(wrapper.find(LoggedInFooter).exists()).toBeFalsy();
  });
});
