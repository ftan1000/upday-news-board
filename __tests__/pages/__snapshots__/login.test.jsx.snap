// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders snapshot 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
>
  <header>
    <div
      className="MuiBox-root css-0"
    >
      <header
        className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
      >
        <div
          className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
        >
          <div
            className="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
          >
            UPDAY News
          </div>
        </div>
      </header>
    </div>
  </header>
  <main>
    <div
      className="MuiBox-root css-binzgt"
    >
      <div
        className="MuiBox-root css-binzgt"
      >
        Log in to access the UPDAY news boards:
      </div>
      <form
        onSubmit={[Function]}
      >
        <div
          aria-label="Enter your email address"
          className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="email"
            id="email-label"
          >
            Enter your email address
            <span
              aria-hidden={true}
              className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="email"
              autoFocus={true}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="email"
              name="email"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="example@email.com"
              required={true}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-173wfxe"
              >
                <span>
                  Enter your email address
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1vhaqj4-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </main>
</main>
`;
